client/state/sites/plans/selectors.js

```diff
@@ -22,13 +23,43 @@ export function getPlansBySiteId( state, siteId ) {
 	return state.sites.plans[ siteId ] || initialSiteState;
 }
 
-export function hasDomainCredit( state, siteId ) {
+export const getCurrentPlan = ( state, siteId ) => {
 	const plans = getPlansBySiteId( state, siteId );
 	if ( plans.data ) {
 		const currentPlan = find( plans.data, 'currentPlan' );
-		return currentPlan.hasDomainCredit;
+
+		if ( currentPlan ) {
+			return currentPlan;
+		}
+
+		const site = getSite( state, siteId );
+
+		if ( ! site.plan ) {
+			return null;
+		}
+
+		// does site enterprise plan?
+		if ( site.plan.product_id === 1007 ) {
+			// lets assign business plan for old enterprise plan
+			return find( plans.data, { productSlug: 'business-bundle' } );
+		}
+
+		return site.plan;
```

Comment:
Why are we duplicating this code here? Is there a way we could just stick with one instance of the logic and call it from the other place? I think here is a good place to have it live, in the state tree…